src/factors/flux/FluxModelsPose2Pose2.jl

```diff
@@ -113,43 +111,60 @@ function sampleFluxModelsPose2Pose2(nfb::FluxModelsPose2Pose2,
 end
 
 # Convenience function to help call the right constuctor
-FluxModelsPose2Pose2( allModels::Vector{P},
-                      jvd::D,
-                      naiveModel::M,
-                      naiveFrac::Real=0.5,
-                      ss::Function=sampleFluxModelsPose2Pose2,
-                      DT::Real=0.0,
-                      shuffle::Bool=true ) where {P, M <: SamplableBelief, D <: AbstractMatrix} = FluxModelsPose2Pose2{P,D,M}(
+FluxModelsPose2Pose2( allModels::Vector,
+                      jvd::AbstractMatrix,
+                      naiveModel::SamplableBelief,
+                      naiveFrac::Float64=0.5,
+                      DT::Float64=0.0,
+                      shuffle::Bool=true ) = FluxModelsPose2Pose2(
                                         allModels,
                                         jvd,
                                         naiveModel,
-                                        naiveFrac,
+                                        Ref(naiveFrac),
                                         Pose2Pose2(MvNormal(zeros(3),diagm(ones(3)))), # this dummy distribution does not get used
-                                        ss,
-                                        DT,
-                                        shuffle )
+                                        Ref(DT),
+                                        Ref(shuffle) )
 #
+# struct FluxModelsPose2Pose2{P,D<:AbstractArray,M<:SamplableBelief} <: AbstractRelativeRoots
+#   allPredModels::Vector{P}
+#   joyVelData::D
+#   naiveModel::M
+#   naiveFrac::Ref{Float64}
+#   # TODO type stability likely wants parameter Pose2Pose2{<:SamplableBelief}
+#   Zij::Pose2Pose2{<:SamplableBelief}
+#   DT::Ref{Float64}
+#   shuffle::Ref{Bool}
+# end
 
 
-function (nfb::FluxModelsPose2Pose2)(
-            res::AbstractArray{<:Real},
-            userdata::FactorMetadata,
-            idx::Int,
-            meas::Tuple{AbstractArray{<:Real},AbstractArray{<:Int},AbstractArray{<:Int}},
-            Xi::AbstractArray{<:Real,2},
-            Xj::AbstractArray{<:Real,2}  )
+function (cfo::CalcFactor{<:FluxModelsPose2Pose2})(meas1,meas2,meas3,Xi,Xj)
+# function (nfb::FluxModelsPose2Pose2)(
+            # res::AbstractArray{<:Real},
+            # userdata::FactorMetadata,
+            # idx::Int,
+            # meas::Tuple{AbstractArray{<:Real},AbstractArray{<:Int},AbstractArray{<:Int}},
+            # Xi::AbstractArray{<:Real,2},
+            # Xj::AbstractArray{<:Real,2}  )
   #
+  userdata = cfo.metadata
+  nfb = cfo.factor
+  fmd = cfo.metadata
+  meas = (meas1, meas2, meas3)
+
+  ## MAYBE JUST MOVE THIS INTO getSample and be done
   # if, use prediction sample
-  if meas[2][idx] == 2
+  if meas[2] == 2 # meas2 == 2
```

Comment:
meas2 (not `meas[2]`)

Reply:
I did this: `meas = (meas1, meas2, meas3)` I wasn't too sure why measurements were no longer coming in as a tuple